llvm/tools/obj2yaml/xcoff2yaml.cpp

```diff
@@ -106,6 +111,79 @@ Error XCOFFDumper::dumpSections(ArrayRef<Shdr> Sections) {
   return Error::success();
 }
 
+static void
+dumpFileAuxSym(std::vector<std::unique_ptr<XCOFFYAML::AuxSymbolEnt>> &AuxEntTbl,
+               XCOFF::CFileStringType Type, StringRef FileStr) {
+  XCOFFYAML::FileAuxEnt FileAuxSym;
+  FileAuxSym.FileNameOrString = FileStr;
+  FileAuxSym.FileStringType = Type;
+  AuxEntTbl.push_back(std::make_unique<XCOFFYAML::FileAuxEnt>(FileAuxSym));
+}
+
+static void
+dumpStatAuxSym(std::vector<std::unique_ptr<XCOFFYAML::AuxSymbolEnt>> &AuxEntTbl,
+               const object::XCOFFSectAuxEntForStat &AuxEntPtr) {
+  XCOFFYAML::SectAuxEntForStat StatAuxSym;
+  StatAuxSym.SectionLength = AuxEntPtr.SectionLength;
+  StatAuxSym.NumberOfLineNum = AuxEntPtr.NumberOfLineNum;
+  StatAuxSym.NumberOfRelocEnt = AuxEntPtr.NumberOfRelocEnt;
+  AuxEntTbl.push_back(
```

Comment:
Thank you for your suggestions! (Sorry, I didn't notice this until after the update).
Although I have no objection about your proposal, actually I'm not sure which implementation is more preferable since current style aligns with other formats like COFF or ELF. And does it make more sense to put the lengthy implementation type code here instead of in the header file?
Hi James @jh7370 , what do you think?

Reply:
I think the implementation of the constructors can be in the cpp, should you choose to go down that route. I think having constructors will likely make things cleaner, but you lose the POD nature of the objects. I've got no strong feeling on whether that is an issue or not, and leave it up to you two and other XCOFF stakeholders to agree on that (I don't think it needs to match COFF/ELF etc specifically).

Reply:
The functionality of the function is to create a new object `SectAuxEntForStat` and initiate with `XCOFFSectAuxEntForStat` , using constructor is object oriented.
it is a suggestion , feel free not to modify if you want to keep current implement.

Reply:
Thank you @diggerlin @jh7370 , well then I would prefer to keep current style.